website/docs/r/kms_grant.html.markdown

```diff
@@ -0,0 +1,75 @@
+---
+layout: "aws"
+page_title: "AWS: aws_kms_grant"
+sidebar_current: "docs-aws-resource-kms-grant"
+description: |-
+  Provides a resource-based access control mechanism for KMS Customer Master Keys.
+---
+
+# aws_kms_grant
+
+Provides a resource-based access control mechanism for a KMS customer master key.
+
+## Example Usage
+
+```hcl
+resource "aws_kms_key" "a" {}
+
+resource "aws_iam_role" "a" {
+name = "iam-role-for-grant"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "lambda.amazonaws.com"
+      },
+      "Effect": "Allow",
+      "Sid": ""
+    }
+  ]
+}
+EOF
+}
+
+resource "aws_kms_grant" "a" {
+  name              = "my-grant"
+  key_id            = "${aws_kms_key.a.key_id}"
+  grantee_principal = "${aws_iam_role.a.arn}"
+  operations        = [ "Encrypt", "Decrypt", "GenerateDataKey" ]
+  constraints {
+    encryption_context_equals {
+      Department = "Finance"
+    }
+  }
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `name` - (Optional, Forces new resources) A friendly name for identifying the grant.
+* `key_id` - (Required, Forces new resources) The unique identifier for the customer master key (CMK) that the grant applies to.
+* `grantee_principal` - (Required, Forces new resources) The principal that is given permission to perform the operations that the grant permits.
```

Comment:
Could we document here the eventual consistency problem and workaround we have in place, with the ARN weirdness?

Reply:
Done.